File: data-structure/js/stack.js
var Stack = /** @class */ (function () {
    function Stack(capacity) {
        this.values = new Array(capacity);
        this.capacity = capacity;
        this.top = -1;
    }
    Stack.prototype.filled = function () {
        return this.top === this.capacity - 1;
    };
    Stack.prototype.empty = function () {
        return this.top === -1;
    };
    Stack.prototype.stack = function (value) {
        if (this.filled())
            console.log("The stack is already filled");
        else {
            this.top++;
            this.values[this.top] = value;
        }
    };
    Stack.prototype.unstack = function () {
        if (this.empty())
            console.log("The stack is empty");
        else {
            this.top--;
        }
    };
    Stack.prototype.getTop = function () {
        if (this.empty())
            console.log("The stack is empty");
        else {
            return this.values[this.top];
        }
    };
    return Stack;
}());
var s = new Stack(5);
s.getTop();
s.stack(1);
s.stack(2);
s.stack(3);
s.stack(4);
s.stack(5);
s.unstack();
s.stack(6);
console.log(s.getTop());
s.unstack();
s.unstack();
s.unstack();
s.unstack();
s.unstack();

File: data-structure/java/Stack.java
public class Stack {
    private int capacity;
    private int top = -1;
    private int values[];

    public Stack(int capacity) {
        this.capacity = capacity;
        this.values = new int[capacity];
    }

    public void print() {
        for (int i = 0; i <= this.top; i++) {
            System.out.println(i + " -- " + this.values[i]);
        }
    }

    public void stack(int value) {
        if (this.top == this.capacity - 1) {
            System.out.println("The array is already filled");
            return;
        }

        this.top += 1;
        this.values[this.top] = value;
    }

    public void unstack() {
        if (this.top == -1) {
            System.out.println("The array is empty");
            return;
        }

        this.top -= 1;
    }

    public int getTop() {
        return this.values[this.top];
    }

    public static void main(String[] args) {
        Stack s = new Stack(5);
        s.stack(5);
        s.stack(3);
        s.stack(1);
        s.stack(4);
        s.stack(2);
        s.print();
        s.unstack();
        s.unstack();
        s.unstack();
        s.unstack();
        s.unstack();
        s.print();
        s.stack(6);
        System.out.println(s.getTop());
    }
}
